scripts/deterministic/DeployScroll.s.sol

```diff
@@ -1033,7 +1041,7 @@ contract DeployScroll is DeterminsticDeployment {
         upgrade(L1_PROXY_ADMIN_ADDR, L1_ERC1155_GATEWAY_PROXY_ADDR, L1_ERC1155_GATEWAY_IMPLEMENTATION_ADDR);
     }
 
-    function deployL1GasTokenGateway() private gasToken(true) {
+    function deployL1GasTokenGateway() private gasToken(true) except(Layer.L2) {
```

Comment:
I don't really understand this, what will go wrong if we simulate this on L2?

Reply:
Yes, that's because here:https://github.com/scroll-tech/scroll-contracts/blob/feat-deterministic-deployment/src/alternative-gas-token/L1GasTokenGateway.sol#L70

in the constructor of L1GasTokenGateway, it will call to get the decimal of gas token. On layer2, there are no gas tokne deployed.